src/napari_experimental/group_layer.py

```diff
@@ -87,22 +72,22 @@ def _check_already_tracking(self, layer_ptr: Layer) -> bool:
         :param layer_ptr: Reference to the Layer to determine is in the
         model.
         """
-        for tracked_layer in self:
-            if (
-                isinstance(tracked_layer, NodeWrappingLayer)
-                and tracked_layer.layer is layer_ptr
-            ):
+        for node in self:
+            if not node.is_group() and node.layer is layer_ptr:
```

Comment:
Should `_check_already_tracking` also look inside nested groups for matching layers? For example, taking the `nested_group_data` in `test_qt_group_layer_model`, you can call `root.add_new_item(layer_ptr=points_aa1)` to add a node with another reference to the same points layer (as the original is nested inside a sub-group)